### PR TITLE
6582 - Incorrect documentation for some string operations

### LINKED ISSUE #6582: Incorrect documentation for some string operations
### Describe the bug

In nushell there seem to be two types of regular expressions, the globs for files ops, and "regular expression" for string operations ; the later are not (to my knowledge) documented, which can lead to surprising results

### How to reproduce

```
> '/home/joe/test.txt' | str ends-with '.txt'
true
> '/home/joe/test.txt' | str ends-with '.txt|.sh'
false
```
The second should be `true` _if the regular expression are the usual type of syntax used in editors / languages_.
The other possibility is that nushell as its own definition, which is not documented. 

### Expected behavior

_if its not a bug_, then somewhere in the book, have event a short chapter describing the expected syntax for regular expressions in string ops, otherwise have the second test return true.

### Screenshots

_No response_

### Configuration

0.68.1
linux

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MichelMunoz**
- fixed documentation as per https://github.com/nushell/nushell/issues/6582#issuecomment-1253850340 (with some adaptation depending on context)
- changed terms in the "search terms" (not sure if there is a limited list of allowed terms here)
- replaced name of field/variable in associated code to avoid future confusion in PR-ers, I used `substring`
- for :
str contains
str ends-with
str start-with
str index-of

closes #6582

### CODE REVIEW

**Comment by fdncred** [comment]
I think these changes are correct however, there is some inconsistency in nushell that we may need to think of, perhaps outside of this PR.

1. operator `=~` is like `str contains` but it's regex
2. operator `starts-with` is like `str starts-with` and calls the rust's `string.starts-with` unless we're talking about dataframes. dataframes uses regex for it's `starts-with` operator
3. operator `ends-with` is like `str ends-with` and calls the rust's `string.ends-with` unless we're talking about dataframes. dataframes uses regex for it's `ends-with` operator

at some point, it may be good to homogenize all these so they work the same way. again, not necessarily in this PR.

File: crates/nu-command/src/strings/str_/contains.rs [hunk 1]
```diff
@@ -17,23 +17,23 @@ impl Command for SubCommand {
...
     }
 
     fn usage(&self) -> &str {
-        "Checks if string contains pattern"
+        "Checks if input contains string"
     }
 
     fn search_terms(&self) -> Vec<&str> {
-        vec!["pattern", "match", "find", "search"]
+        vec!["contain", "match", "find", "search"]
```
Comment (sholderbach) [inline crates/nu-command/src/strings/str_/contains.rs:36]:
```suggestion
        vec!["substring", "match", "find", "search"]
```

File: crates/nu-command/src/strings/str_/ends_with.rs [hunk 2]
```diff
@@ -26,11 +26,11 @@ impl Command for SubCommand {
     }
 
     fn usage(&self) -> &str {
-        "Check if a string ends with a pattern"
+        "Check if an input ends with a string"
     }
 
     fn search_terms(&self) -> Vec<&str> {
-        vec!["pattern", "match", "find", "search"]
+        vec!["end", "match", "find", "search"]
```
Comment (sholderbach) [inline crates/nu-command/src/strings/str_/ends_with.rs:33]:
```suggestion
        vec!["suffix", "match", "find", "search"]
```

File: crates/nu-command/src/strings/str_/index_of.rs [hunk 3]
```diff
@@ -28,31 +28,31 @@ impl Command for SubCommand {
...
     }
 
     fn usage(&self) -> &str {
-        "Returns start index of first occurrence of pattern in string, or -1 if no match"
+        "Returns start index of first occurrence of string in input, or -1 if no match"
     }
 
     fn search_terms(&self) -> Vec<&str> {
-        vec!["pattern", "match", "find", "search"]
+        vec!["index", "match", "find", "search"]
```
Comment (sholderbach) [inline crates/nu-command/src/strings/str_/index_of.rs:55]:
```suggestion
        vec!["match", "find", "search"]
```

File: crates/nu-command/src/strings/str_/starts_with.rs [hunk 4]
```diff
@@ -33,11 +33,11 @@ impl Command for SubCommand {
     }
 
     fn usage(&self) -> &str {
-        "Check if string starts with a pattern"
+        "Check if an input starts with a string"
     }
 
     fn search_terms(&self) -> Vec<&str> {
-        vec!["pattern", "match", "find", "search"]
+        vec!["start", "match", "find", "search"]
```
Comment (sholderbach) [inline crates/nu-command/src/strings/str_/starts_with.rs:40]:
```suggestion
        vec!["prefix", "match", "find", "search"]
```

**Review by sholderbach — COMMENTED** [review]
Thanks for taking care of that! Can you run `cargo fmt --all` on the project? If you don't have a full Rust development environment set up we can take care of that as well.

We also try to only use search terms that are not already found by the command name. The test from #6380 is the only thing that seems to fail.

**Comment by MichelMunoz** [comment]
> Thanks for taking care of that! Can you run cargo fmt --all on the project? If you don't have a full Rust development environment set up we can take care of that as well.
I don't have a rust environment (I just used a basic text editor for the changes, but normally I did not change any indentation or end of line)

**Comment by sholderbach** [comment]
No problem, can quickly run that through `rustfmt`. I don't have the wrapping heuristics for long parameter lists in my head.

**Pull request merged**